### ISSUE TITLE
Difficulty multipliers do not apply in Multiplayer

### ISSUE THREAD

**Issue by ZenoG2** [ppy/osu#29313]
### Type

Game behaviour

### Bug description

The difficulty shown is not affected by mod multiplier in multiplayer, unlike stable.
Original post on forums before confirming this is a bug: https://osu.ppy.sh/community/forums/topics/1956669?n=1

### Screenshots or videos

![image](https://github.com/user-attachments/assets/1136e064-2919-4a2a-ade8-9a2250dc9b80)

![image](https://github.com/user-attachments/assets/768c4f45-1d6a-4362-bd15-2778a2fe305f)

![image](https://github.com/user-attachments/assets/2783572a-0801-4a67-98e2-33a1ec90bd3e)

### Version

2024.731.0-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/16512801/compressed-logs.zip)

**Comment by yesseruser** [comment]
The incorrect rating also shows in the room's tile:
![image](https://github.com/user-attachments/assets/21885614-4aab-4e62-8193-c5d8f7f71612)

**Comment by ZenoG2** [comment]
This issue also applies to "difficulty range" in singleplayer.

https://github.com/user-attachments/assets/b06deb85-6a72-4a2e-b0c7-fddf5a85245b

**Comment by bdach** [comment]
@ZenoG2 that's a separate issue, separate issue thread please

**Comment by bdach** [comment]
This one is sorta difficult to properly deal with because what happens when the user doesn't have the beatmap locally?

I can see two options here:

- somehow get web to return the actual effective star rating including mods
- fix this client-side ourselves, by requesting difficulty attributes from API and doing a local diffcalc

@ppy/team-web any thoughts on this?

**Comment by nanaya** [comment]
if the data is somewhere in the database then sure, osu-web can pass it along. Last I remember it's only for stable client related mods though? Or have lazer mods also been added?

**Comment by bdach** [comment]
Yeah nothing has changed on that front, database only contains the attributes for mods that exist in stable.

I wonder if at some point we want `osu-rank-lookup-cache` to support lazer mods. Maybe that'd be a start to all of this.

**Comment by smoogipoo** [comment]
> I wonder if at some point we want `osu-rank-lookup-cache` to support lazer mods.

Unless I'm missing a reason, we already have [`osu-beatmap-difficulty-lookup-cache`](https://github.com/ppy/osu-beatmap-difficulty-lookup-cache) that's intended for this purpose. I believe it's being forced to use only [databased attributes](https://github.com/ppy/osu-beatmap-difficulty-lookup-cache/blob/db2203368221109803f2031788da31deb94e0f11/BeatmapDifficultyLookupCache/DifficultyCache.cs#L46) because realtime diffcalc was too expensive (and is why I'm worried about https://github.com/ppy/osu-queue-score-statistics/pull/274) so it doesn't support lazer mods, but it looks like even that would be enough for the OP.

There already is an endpoint for the above somewhere inside osu-web, so perhaps that can be reused.

**Comment by shinmorisawa** [comment]
i have noticed this for a long time.

**Comment by Qunas101** [comment]
Want to chime in and note that this also happens to the new freestyle icon for converted maps, i.e. your friend that the game says is playing an 8* mania map is actually playing a 3*

<details>

<summary>Spoiler</summary>

![Image](https://github.com/user-attachments/assets/10ec69ef-3c9b-44ef-92ef-7286cfc33731)


![Image](https://github.com/user-attachments/assets/35d2705c-dc51-4dc7-8df6-6395bc94c4af)

</details>
